--- links-1.04/https.c.orig	2018-02-08 19:01:11.000000000 +0100
+++ links-1.04/https.c	2020-09-06 10:00:32.673251769 +0200
@@ -53,7 +53,11 @@ SSL *getSSL(void)
 			RAND_add(pool, sizeof pool, sizeof pool);
 		}
 #endif
+#if !(OPENSSL_API_COMPAT < 0x10100000L)
+		OPENSSL_init_ssl(0, NULL);
+#else
 		SSLeay_add_ssl_algorithms();
+#endif
 		context = SSL_CTX_new(SSLv23_client_method());
 		SSL_CTX_set_options(context, SSL_OP_ALL);
 		SSL_CTX_set_default_verify_paths(context);
